src/DataStructures/DataBox/Tag.hpp

```diff
@@ -155,4 +155,34 @@ struct PrefixTag {};
  * \see DataBoxGroup SimpleTag
  */
 struct ComputeTag {};
+
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Mark a struct as a reference tag by inheriting from this.
+ *
+ * \details
+ * A reference tag is used to identify an item in a DataBox that is a const
+ * reference to a subitem of a tagged container (such as a Variables or
+ * GlobalCache) in the DataBox
+ *
+ * \derivedrequires
+ * - type alias `base` that is the simple tag from which it is derived (and is
+ *   the tag used to fetch the subitem from the tagged container)
+ * - type alias `parent_tag` that is the tag for the tagged container
```

Comment:
Will the reference tags also work without the requirement that the reference has to point into a tagged container? That was my impression when you mentioned it in Slack a while ago. Here's what we talked about back then, copying it here so I don't have to paraphrase the question and answer:

> Different question: The analytic-data class for elasticity systems provides the constitutive relation for the solve (just like how hydro data can provide an EOS created from options). The analytic-data object is stored in the const global cache. Now I’d like to retrieve the constitutive relation from the DataBox directly with `db:get<Elasticity::Tags::ConstitutiveRelation<Dim>>(box)` , but would like to avoid copying the object over from the analytic-data object into another tag. Any ideas how this could reasonably work?

> (...) you define a new tag derived from `db::ReferenceTag` and the simple tag `Elasticity::Tags::ConstitutiveRelation<Dim>` templated on the parent tag, which in your case would be the tag for the analytic-data object, and define `static const auto& get(const typename parent_tag::type& parent_value)` that returns a reference to the constitutive relation given the analytic-data object…

Reply:
yes, I think that should work; I'll write a test (which can't appear in this PR, but in a future one) to make sure it works correctly...

Reply:
@nilsleiffischer I've update the documentation; please take a look